src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1015,6 +1015,56 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
                 throw new DaprException("State operation failed: the Dapr endpoint indicated a failure. See InnerException for details.", ex);
             }
         }
+
+        /// <inheritdoc/>
+        public async override Task<StateQueryResponse<TValue>> QueryStateAsync<TValue>(StateQueryRequest request, CancellationToken cancellationToken = default)
+        {
+            var queryRequest = new Autogenerated.QueryStateRequest()
+            {
+                StoreName = request.StoreName,
+                Query = request.Query
+            };
+
+            if (request.Metadata != null)
+            {
+                foreach (var kvp in request.Metadata)
+                {
+                    queryRequest.Metadata.Add(kvp.Key, kvp.Value);
+                }
+            }
+
+            var options = CreateCallOptions(headers: null, cancellationToken);
+
+            try
+            {
+                var queryResponse = await client.QueryStateAlpha1Async(queryRequest, options);
+                var items = queryResponse.Results.Select(res =>
+                    new StateQueryItem<TValue>()
+                    {
+                        Key = res.Key,
+                        Data = TypeConverters.FromJsonByteString<TValue>(res.Data, this.JsonSerializerOptions),
+                        ETag = res.Etag,
+                        Error = res.Error,
```

Comment:
The fact that the error property exists means that we can't **just** try to deserialize every item, we have to check for an error first.

Reply:
Yeah I'll add that check in as part of the solution we discussed below.